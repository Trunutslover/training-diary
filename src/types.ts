export enum actionTypes {
    TOGGLE_IS_DONE = 'TOGGLE_IS_DONE',
    ADD_SET = 'ADD_SET',
    REMOVE_SET = 'REMOVE_SET',
    SET_REPS = 'SET_REPS',
    SET_EXERCISE_NAME = 'SET_EXERCISE_NAME',
    ADD_EXERCISE = 'ADD_EXERCISE',
    REMOVE_EXERCISE = 'REMOVE_EXERCISE',
    SET_AUTH = 'SET_AUTH',
    REMOVE_AUTH = 'REMOVE_AUTH'
}

export interface actionInterface {
    type: actionTypes,
    payload?: any
}

export interface ExerciseInterface {
    number: number,
    name: string,
    sets: {
        reps: number,
        weight: number
    }[]
}

export interface TrainingInterface {
    id: number,
    isDone: boolean,
    exercises: ExerciseInterface[]
}

export interface TrainingsInterface {
    id: number
}

export interface ProfileInterface {
    name: string,
    avatar: string | undefined,
    age: number,
    sex: boolean
}