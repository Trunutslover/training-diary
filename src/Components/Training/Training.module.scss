@import "../../constants";
@import "../../mixins";

.Training {
  display: flex;
  flex-direction: column;
  padding: 10px 20px;
  box-sizing: border-box;

  color: $color-secondary-b-2;
  background-color: $color-secondary-b-4;
}

.done {
  color: $color-secondary-a-2;
  background-color: $color-secondary-a-4;
}

.button {
  @include redButton;

  align-self: center;
}

.buttonDone {
  @include greenButton;

  align-self: center;
}

.title {
  margin: 0;
  margin-bottom: 5px;
}