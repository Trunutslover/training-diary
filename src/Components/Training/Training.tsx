import React from 'react';
import {actionInterface, ExerciseInterface} from "../../types";

interface Props {
    id: number,
    isDone: boolean,
    exercises: ExerciseInterface[],
    toggleIsDoneAC: (id: number) => actionInterface
}

const Training = ({id, isDone, exercises, toggleIsDoneAC}: Props):React.ReactElement => {
    return (
        <section>
            <p>
                Training number {id}
            </p>
            <p>
                Training is done: {isDone ? `yes` : `no`}
            </p>
            <p>
                Exercises: {exercises.map(value => <span key={value.number}>number - {value.number} name - {value.name} sets - {value.sets.map((value, index) => <span key={index}>{value} </span>)}</span>)}
            </p>
            {isDone ? <button onClick={() => toggleIsDoneAC(id)}>Training is not done</button> : <button onClick={() => toggleIsDoneAC(id)}>Training is done</button>}
        </section>
    )
};

export default Training;