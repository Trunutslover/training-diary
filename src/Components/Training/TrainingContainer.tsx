import React from 'react';
import {
    addExerciseAC,
    addSetAC, removeExerciseAC,
    removeSetAC,
    setExerciseNameAC,
    setRepsAC, setTimestampAC,
    toggleIsDoneAC
} from "../../redux/trainingReducer";
import {compose} from "redux";
import {connect} from "react-redux";
import Training from "./Training";
import {actionInterface, TrainingInterface} from "../../types";
import {useRouteMatch} from "react-router";

interface Props {
    training: TrainingInterface,
    toggleIsDoneAC: () => actionInterface,
    addSetAC: (exerciseNumber: number) => actionInterface,
    removeSetAC: (exerciseNumber: number, setNumber: number) => actionInterface,
    setRepsAC: (exerciseNumber: number, value: string, setNumber?: number, isReps?: boolean) => actionInterface,
    setExerciseNameAC: (exerciseNumber: number, value: string) => actionInterface,
    addExerciseAC: () => actionInterface,
    removeExerciseAC: (exerciseNumber: number) => actionInterface,
    setTimestampAC: (timestamp: number) => actionInterface
}

const TrainingContainer = (props: Props): React.ReactElement => {
    const match: any = useRouteMatch('/trainings/:trainingId');

    if(parseInt(match.params.trainingId) === props.training.id) {
        return (
            <Training
                id={props.training.id}
                timestamp={props.training.timestamp}
                isDone={props.training.isDone}
                exercises={props.training.exercises}
                toggleIsDoneAC={props.toggleIsDoneAC}
                addSetAC={props.addSetAC}
                removeSetAC={props.removeSetAC}
                setRepsAC={props.setRepsAC}
                setExerciseNameAC={props.setExerciseNameAC}
                addExerciseAC={props.addExerciseAC}
                removeExerciseAC={props.removeExerciseAC}
                setTimestampAC={props.setTimestampAC}
            />
        )
    } else {
        return <div>Preloader</div>
    }

};

const mapStateToProps = (state: any) => {
    return {
        training: state.training
    }
};

const mapDispatchToProps = {
    toggleIsDoneAC,
    addSetAC,
    removeSetAC,
    setRepsAC,
    setExerciseNameAC,
    addExerciseAC,
    removeExerciseAC,
    setTimestampAC
};

export default compose(
    connect(mapStateToProps, mapDispatchToProps)
)(TrainingContainer);