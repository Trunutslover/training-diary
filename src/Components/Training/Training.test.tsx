import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Training from "./Training";
import {actionTypes} from "../../types";
import {shallowToJson} from "enzyme-to-json";

Enzyme.configure({adapter: new Adapter()});

describe(`Training snapshot`, () => {
    it('Training should render correctly', () => {
        const mockProps = {
            id: 1,
            isDone: false,
            exercises: [
                {
                    number: 1,
                    name: `squats`,
                    sets: [10, 12, 12, 10, 8]
                }
            ]
        };

        const output = shallow(<Training id={mockProps.id} isDone={mockProps.isDone} exercises={mockProps.exercises} toggleIsDoneAC={() => ({type: actionTypes.TOGGLE_IS_DONE})}/>);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});