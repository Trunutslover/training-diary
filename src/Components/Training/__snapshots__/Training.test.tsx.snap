// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Training snapshot Training should render correctly 1`] = `
<section>
  <p>
    Training number 
    1
  </p>
  <p>
    Training is done: 
    no
  </p>
  <p>
    Exercises: 
    <span
      key="1"
    >
      number - 
      1
       name - 
      squats
       sets - 
      <span
        key="0"
      >
        10
         
      </span>
      <span
        key="1"
      >
        12
         
      </span>
      <span
        key="2"
      >
        12
         
      </span>
      <span
        key="3"
      >
        10
         
      </span>
      <span
        key="4"
      >
        8
         
      </span>
    </span>
  </p>
  <button
    onClick={[Function]}
  >
    Training is done
  </button>
</section>
`;
